src/Compilers/CSharp/Test/Emit3/Semantics/CollectionExpressionArgumentTests.cs

```diff
@@ -3997,6 +4101,1353 @@ static void Main()
+                      IL_0016:  callvirt   "void System.Collections.Generic.List<T>.Add(T)"
+                      IL_001b:  ret
+                    }
+                    """;
+                verifier.VerifyIL("Program.Create1<T>", expectedIL);
+                verifier.VerifyIL("Program.Create2<T>", expectedIL);
+            }
+            else
+            {
+                comp.VerifyEmitDiagnostics(
+                    // (11,16): error CS1501: No overload for method '<signature>' takes 1 arguments
+                    //         return [with(c), x, y, z];
+                    Diagnostic(ErrorCode.ERR_BadArgCount, "[with(c), x, y, z]").WithArguments("<signature>", "1").WithLocation(11, 16),
+                    // (15,22): error CS1739: The best overload for '<signature>' does not have a parameter named 'capacity'
+                    //         return [with(capacity: c), x, y, z];
+                    Diagnostic(ErrorCode.ERR_BadNamedArgument, "capacity").WithArguments("<signature>", "capacity").WithLocation(15, 22));
+            }
+
+            string sourceC = $$"""
+                using System.Collections.Generic;
+                class Program
+                {
+                    static {{typeName}}<T> Create1<T>(IEnumerable<T> c, T x, T y, T z)
+                    {
+                        return [with(c), x, y, z];
+                    }
+                    static {{typeName}}<T> Create2<T>(IEnumerable<T> c, T x, T y, T z)
+                    {
+                        return [with(collection: c), x, y, z];
+                    }
+                }
+                """;
+            comp = CreateCompilation(sourceC);
+            if (isMutable)
+            {
+                comp.VerifyEmitDiagnostics(
+                    // (6,22): error CS1503: Argument 1: cannot convert from 'System.Collections.Generic.IEnumerable<T>' to 'int'
+                    //         return [with(c), x, y, z];
+                    Diagnostic(ErrorCode.ERR_BadArgType, "c").WithArguments("1", "System.Collections.Generic.IEnumerable<T>", "int").WithLocation(6, 22),
+                    // (10,22): error CS1739: The best overload for '<signature>' does not have a parameter named 'collection'
+                    //         return [with(collection: c), x, y, z];
+                    Diagnostic(ErrorCode.ERR_BadNamedArgument, "collection").WithArguments("<signature>", "collection").WithLocation(10, 22));
+            }
+            else
+            {
+                comp.VerifyEmitDiagnostics(
+                    // (6,16): error CS1501: No overload for method '<signature>' takes 1 arguments
+                    //         return [with(c), x, y, z];
+                    Diagnostic(ErrorCode.ERR_BadArgCount, "[with(c), x, y, z]").WithArguments("<signature>", "1").WithLocation(6, 16),
+                    // (10,22): error CS1739: The best overload for '<signature>' does not have a parameter named 'collection'
```

Comment:
>The best overload for '<signature>' does not have a parameter named 'collection' [](http://example.com/codeflow?start=46&length=80)

Consider leaving a follow-up comment on diagnostic quality

Reply:
Is the concern with the name of the signature or the error in general? Any thoughts on how this should be reported?

Reply:
One option is to use a dedicated error message "The best overload for the collection constructor does not have a parameter named 'collection'."

Reply:
https://github.com/dotnet/roslyn/issues/78568